src/buildercore/core.py

```diff
@@ -85,6 +88,24 @@ def _set_raw_subscription_attribute(sns_connection, subscription_arn):
 
 sns.connection.SNSConnection.set_raw_subscription_attribute = _set_raw_subscription_attribute
 
+def all_sns_subscriptions(region, stackname=None):
+    """returns all subscriptions to all sns topics.
+    optionally filtered by subscription endpoints matching given stack"""
+    conn = boto_sns_conn(region)
+    response = conn.get_all_subscriptions() # TODO: needs pagination!
```

Comment:
shouldn't boto (or boto3) do pagination for you?

Reply:
(perhaps with other Python functions provided, not sure)

Reply:
boto3 does pagination for you at a certain level. I've never seen boto2 do pagination for me